presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java

```diff
@@ -387,14 +387,13 @@ private boolean tryRehash()
 
         // An estimate of how much extra memory is needed before we can go ahead and expand the hash table.
         // This includes the new capacity for groupAddressByHash, rawHashByHashPosition, groupIdsByHash, and groupAddressByGroupId as well as the size of the current page
-        preallocatedMemoryInBytes = (newCapacity - hashCapacity) * (long) (Long.BYTES + Integer.BYTES + Byte.BYTES) +
-                (calculateMaxFill(newCapacity) - maxFill) * Long.BYTES +
+        preallocatedMemoryInBytes = newCapacity * (long) (Long.BYTES + Integer.BYTES + Byte.BYTES) +
```

Comment:
do we know the reason why we removing the `hashCapacity` value from here?

Reply:
The rehash process works like this: 
1. Allocate a new table with `newCapacity` (it is double the size of the original capacity)
2. Rehash the data in the old table into the new table.

During the rehash process, there would be a moment that we would keep both of the new table and the old table, therefore we should account for both of them.

Previously we account for the only size difference between the new table and the old table, i.e. `(newCapacity - hashCapacitty)` as the memory that we need to allocate. Now we need to account for the complete new table. Therefore it is `newCapacity` now.